momento-sdk/src/main/java/momento/sdk/ScsClient.java

```diff
@@ -19,11 +19,27 @@ abstract class ScsClient implements AutoCloseable {
 
   private static final Metadata.Key<String> CACHE_NAME_KEY =
       Metadata.Key.of("cache", ASCII_STRING_MARSHALLER);
+  private static final Metadata.Key<String> SDK_AGENT_KEY =
+      Metadata.Key.of("Agent", ASCII_STRING_MARSHALLER);
+  private static final Metadata.Key<String> RUNTIME_VERSION_KEY =
+      Metadata.Key.of("Runtime-Version", ASCII_STRING_MARSHALLER);
+  private boolean hasSentOneTimeHeaders = false;
 
   protected Metadata metadataWithCache(String cacheName) {
-    final Metadata metadata = new Metadata();
-    metadata.put(CACHE_NAME_KEY, cacheName);
+    if (hasSentOneTimeHeaders) {
+      Metadata metadata = new Metadata();
+      metadata.put(CACHE_NAME_KEY, cacheName);
+      return metadata;
+    }
+    hasSentOneTimeHeaders = true;
+    String sdkVersion =
+        String.format("java:%s", this.getClass().getPackage().getImplementationVersion());
+    String runtimeVer = System.getProperty("java.version");
```

Comment:
Is it worth including the vendor `System.getProperty("java.vendor")` as well, so we know which jdk is being used? The non-android runtime version string I'm using for kotlin at the moment is `val runtimeVersion = "$javaVendor-$javaVersion, Kotlin $kotlinVersion"`
cc @cprice404